crossdock/server/server.py

```diff
@@ -105,104 +127,93 @@ class UnknownTransportException(Exception):
 
 class Server(object):
     def __init__(self):
-        self.tracer = Tracer("python", None, ConstSampler(decision=True))
-        opentracing.tracer = self.tracer
+        self.tracer = opentracing.tracer
 
+    @tornado.gen.coroutine
     def start_trace(self, request, response_writer):
-        sstr = serializer.start_trace_request_from_json(request.body)
+        start_trace_req = serializer.start_trace_request_from_json(request.body)
 
         def update_span(span):
-            span.set_baggage_item(constants.baggage_key, sstr.baggage)
-            span.set_tag(ext_tags.SAMPLING_PRIORITY, sstr.sampled)
+            span.set_baggage_item(constants.baggage_key, start_trace_req.baggage)
+            span.set_tag(ext_tags.SAMPLING_PRIORITY, start_trace_req.sampled)
 
-        self.handle_trace_request(request, sstr, update_span, response_writer)
+        response = yield self.handle_downstream_request(request, start_trace_req.downstream,
+                                                        update_span)
+        response_writer.write(response)
+        response_writer.finish()
 
+    @tornado.gen.coroutine
     def join_trace(self, request, response_writer):
+        join_trace_request = serializer.join_trace_request_from_json(request.body) \
+            if request.body else None
 
-        jtr = serializer.join_trace_request_from_json(request.body) if request.body else None
-
-        self.handle_trace_request(request, jtr, None, response_writer)
+        response = yield self.handle_downstream_request(request, join_trace_request.downstream,
+                                                        None)
+        response_writer.write(response)
+        response_writer.finish()
 
-    def handle_trace_request(self, http_request, trace_request, span_handler, response_writer):
+    @tornado.gen.coroutine
+    def handle_downstream_request(self, http_request, downstream, span_handler):
```

Comment:
nit: s/span_handler/update_span_func/

every time I read this code, I stumble on it and have to look back what it means. Giving it a clear meaningful name makes the code much easier to read.